File: 2022-2023/Kurs_programowania/Lista 3/Zadanie 1/Prostokat.java
public class Prostokat extends Czworokat{
    public Prostokat(double bok1,double bok2){
        super("prostokat",bok1,bok2,bok1,bok2);
    }
    public double obliczPole(){
        return bok1*bok2;
    }
    public double obliczObwod(){
        return bok1+bok2+bok3+bok4;
    }
}

File: 2022-2023/Kurs_programowania/Lista 3/Zadanie 2/Prostokat.cpp

#include "Prostokat.hpp"

Prostokat::Prostokat(double bok1, double bok2, double bok3, double bok4, double kat): Czworokat(bok1,bok2,bok3,bok4,kat) {};

double Prostokat::pole() {
    return bok1*bok2;
}

double Prostokat::obwod() {
    return bok1*2+bok2*2;
}

std::string Prostokat::nazwa() {
    return "Prostokat";
}